crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1438,6 +1441,18 @@ impl<'db> ClassType<'db> {
             })
     }
 
+    pub fn is_subclass_of(self, db: &'db dyn Db, other: ClassType) -> bool {
+        // TODO: we need to iterate over the *MRO* here, not the bases
+        (other == self)
+            || self.bases(db).any(|base| match base {
+                Type::Class(base_class) => base_class == other,
+                // `is_subclass_of` is checking the subtype relation, in which gradual types do not
+                // participate, so we should not return `True` if we find `Any/Unknown` in the
+                // bases.
+                _ => false,
```

Comment:
I know you don't like default cases in match statements, but since this is all subject to change soon anyway when we add proper MRO support, I didn't feel it was worth listing out every `Type` variant here.